service/controller/resource/promxy/types.go

```diff
@@ -0,0 +1,52 @@
+package promxy
+
+import (
+	"gopkg.in/yaml.v2"
+)
+
+type Promxy struct {
+	Promxy PromxyConfig `yaml:"promxy"`
+}
+
+// PromxyConfig is the configuration for Promxy itself
+type PromxyConfig struct {
```

Comment:
This can be replaced with https://godoc.org/github.com/jacksontj/promxy/pkg/config#Config

Reply:
Sadly,no, as promxy is using an old prometheus dependency on 1.8 so the version we get by adding this to the project makes us need to acually import prometheus 1.8 instead of 2.20. Morever, the prometheus version is a fork made jacksontj/prometheus

Reply:
This is the part I really do not like

Reply:
Can we at least please add a comment to tell where this comes from.
Maybe using replace in go.mod would do the trick of using the right dependency for us. It's really not good to have to copy all this code in here.

Reply:
Where is the original promxy then ?

Reply:
Beautiful thanks :heart: